systemtest/src/main/java/io/strimzi/systemtest/resources/operator/OlmResource.java

```diff
@@ -6,71 +6,260 @@
+                }
+            }
+        }
+        if (!(CLOSED_MAP_INSTALL_PLAN.keySet().size() > 0)) {
+            throw new RuntimeException("No install plans located in namespace:" + cmdKubeClient().namespace());
+        }
+    }
+
+    /**
+     * Get specific version of cluster operator with prefix name in format: 'strimzi-cluster-operator.v0.18.0'
+     * @return version with prefix name
+     */
+    public static String getClusterOperatorVersion() {
+        String installPlansPureString = cmdKubeClient().exec("get", "installplan").out();
+        String[] installPlansLines = installPlansPureString.split("\n");
+
+        for (String line : installPlansLines) {
+            // line = NAME   CSV   APPROVAL   APPROVED
+            String[] wholeLine = line.split("   ");
+
+            // non-used install plan
+            if (wholeLine[0].equals(getNonUsedInstallPlan())) {
+                return wholeLine[1];
+            }
+        }
+        throw new RuntimeException("Version was not found in the install plan.");
+    }
+
+    public static boolean isUpgradeable() {
+        return !getNonUsedInstallPlan().equals(NO_MORE_NON_USED_INSTALL_PLANS);
+    }
+
+    public static String getNonUsedInstallPlan() {
+        String[] nonUsedInstallPlan = new String[1];
+
+        for (Map.Entry<String, Boolean> entry : CLOSED_MAP_INSTALL_PLAN.entrySet()) {
+            // if value is FALSE we are gonna use it = non-used install plan
+            if (!entry.getValue()) {
+                nonUsedInstallPlan[0] = entry.getKey();
+                break;
+            }
+            nonUsedInstallPlan[0] = NO_MORE_NON_USED_INSTALL_PLANS;
+        }
+
+        LOGGER.info("Non-used install plan is {}", nonUsedInstallPlan[0]);
+        return nonUsedInstallPlan[0];
+    }
+
+    /**
+     * Patches specific non used install plan, which will approve installation. Only for manual installation strategy.
+     * Also updates closedMapInstallPlan map and set specific install plan to true.
+     */
+    private static void approveNonUsedInstallPlan() {
+        String nonUsedInstallPlan = getNonUsedInstallPlan();
+
+        try {
+            LOGGER.info("Approving {} install plan", nonUsedInstallPlan);
+            String dynamicScriptContent =
+                "#!/bin/bash\n" +
+                    cmdKubeClient().cmd() +
+                    " patch installplan " + nonUsedInstallPlan + " --type json  --patch '[{\"op\": \"add\", \"path\": \"/spec/approved\", \"value\": true}]'";
+
+            InputStream inputStream = new ByteArrayInputStream(dynamicScriptContent.getBytes(Charset.defaultCharset()));
+            File patchScript = File.createTempFile("installplan_patch",  ".sh");
+            Files.copy(inputStream, patchScript.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+            Exec.exec("bash", patchScript.getAbsolutePath());
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Comment:
Shouldn't it fail?

Reply:
yeah, I replaced this with `throw new RuntimeException(e)`